tw-tasks-core/src/main/java/com/transferwise/tasks/dao/JdbcTaskDao.java

```diff
@@ -272,13 +279,16 @@ public InsertTaskResponse insertTask(InsertTaskRequest request) {
         DataSourceUtils.releaseConnection(con, dataSource);
       }
 
-      byte[] data = request.getData();
-      if (data != null) {
-        SerializedData serializedData = taskDataSerializer.serialize(data, request.getCompression());
-        jdbcTemplate.update(insertTaskDataSql, args(taskId, Integer.valueOf(serializedData.getDataFormat()), serializedData.getData()));
-        coreMetricsTemplate.registerDaoTaskDataSerialization(request.getType(), data.length, serializedData.getData().length);
-      }
-
+      byte[] data = request.getData() == null ? NULL_BLOB : request.getData();
```

Comment:
We should avoid this block, if both data and context are null / empty.
Just to avoid one more record in database for lightweight tasks.